make/modules/java.base/gensrc/GensrcMisc.gmk

```diff
@@ -50,11 +50,33 @@ $(eval $(call SetupTextFileProcessing, BUILD_VERSION_JAVA, \
         @@VENDOR_URL_VM_BUG@@ => $(VENDOR_URL_VM_BUG), \
 ))
 
+
+# Normalize OPENJDK_TARGET_CPU name to match jdk.internal.util.Architecture enum
+ifneq ($(filter $(OPENJDK_TARGET_CPU), ppc64le), )
+  OPENJDK_TARGET_ARCH_CANONICAL = ppc64
+else ifneq ($(filter $(OPENJDK_TARGET_CPU), s390x), )
+  OPENJDK_TARGET_ARCH_CANONICAL = s390
+else ifneq ($(filter $(OPENJDK_TARGET_CPU), x86_64 amd64), )
+  OPENJDK_TARGET_ARCH_CANONICAL = x64
+else
+  OPENJDK_TARGET_ARCH_CANONICAL := $(OPENJDK_TARGET_CPU)
+endif
+
+# Normalize OPENJDK_TARGET_OS operating system name to match jdk.internal.util.OperatingSystem enum
+ifeq ($(OPENJDK_TARGET_OS), macosx)
+  OPENJDK_TARGET_OS_CANONICAL = macos
+else
+  OPENJDK_TARGET_OS_CANONICAL := $(OPENJDK_TARGET_OS)
+endif
+
 $(eval $(call SetupTextFileProcessing, BUILD_PLATFORMPROPERTIES_JAVA, \
```

Comment:
@erikj79 Is there a better/good way to do these mappings, or select "local" variable names?

Reply:
Not sure if it's better, but you could consider doing this directly in the switch statements in `make/autoconf/platform.m4` and add the corresponding variables in `spec.gmk.in`. That would make them available globally in the build, which may also be detrimental as parts of the build could start relying on them, and we end up with even more variants sprinkled around. I think what you have here is better for now.

Reply:
ok, I'd rather keep it local.  Thanks